src/plugins/ucx/ucx_backend.cpp

```diff
@@ -735,32 +785,40 @@ nixl_status_t
 nixlUcxEngine::internalMDHelper (const nixl_blob_t &blob,
                                  const std::string &agent,
                                  nixlBackendMD* &output) {
-    nixlUcxConnection conn;
     nixlUcxPublicMetadata *md = new nixlUcxPublicMetadata;
-     size_t size = blob.size();
+    size_t size = blob.size();
 
     auto search = remoteConnMap.find(agent);
 
     if(search == remoteConnMap.end()) {
         //TODO: err: remote connection not found
         return NIXL_ERR_NOT_FOUND;
     }
-    conn = (nixlUcxConnection) search->second;
+
+    nixlUcxConnection &conn = search->second;
 
     //directly copy underlying conn struct
     md->conn = conn;
 
-    char *addr = new char[size];
-    nixlSerDes::_stringToBytes(addr, blob, size);
+    std::vector<char> addr(size);
+    nixlSerDes::_stringToBytes(addr.data(), blob, size);
 
-    int ret = uw->rkeyImport(conn.ep, addr, size, md->rkey);
-    if (ret) {
-        // TODO: error out. Should we indicate which desc failed or unroll everything prior
+    bool error = false;
+    for (size_t worker_id = 0; worker_id < uws.size(); worker_id++) {
+        nixlUcxRkey rkey;
+        error = md->conn.getEp(worker_id).rkeyImport(addr.data(), size, rkey);
+        if (error)
+            // TODO: error out. Should we indicate which desc failed or unroll everything prior
+            break;
+        md->rkeys.push_back(rkey);
+    }
+    if (error) {
+        for (size_t worker_id = 0; worker_id < md->rkeys.size(); worker_id++)
```

Comment:
my 5 cents is that for such local iterators that are mentioned multiple times in the expr. below, it's better to use something short, like `wid` or `wId`

Reply:
Makes sense, done.